src/Hypergrid.js

```diff
@@ -1151,7 +1151,7 @@ var Hypergrid = Base.extend('Hypergrid', {
      * @returns {boolean} The given row is fully visible.
      */
     isDataRowVisible: function(r) {
-        return this.renderer.isRowVisible(r);
+        return this.renderer.getVisibleDataRow(r);
```

Comment:
To be honest, I don't really remember why I've changed this (probably was a first attempt to fix it...) and why I didn't use the `this.renderer.isVisibleDataRow(r)` instead.
I've reverted this code and tested. Keyboard navigation, selection and the method to focus in one particular cell look good.